### DISCUSSION TITLE
設定したいDjangoの初期設定はありますか？

### DISCUSSION THREAD

**Discussion by s1-haya** [42-pong/42-pong#38] [Ideas]
### 背景
https://github.com/42-pong/42-pong/issues/32 で他にも初期設定するのがあるのかなと思ったため

### 現状
Djangoの初期設定
- superuserの作成
- timezoneを日本時間（JST）に変更

### 質問
Djangoの初期設定を上記以外にしときたいのがあれば教えて欲しいです。

個人的に必要なのかなと思った初期設定
- データベースの設定をsqliteから[PostgreSQL](https://www.postgresql.org/)にする？
- [INSTALLED_APPS](https://docs.djangoproject.com/en/5.1/ref/settings/#std-setting-INSTALLED_APPS)にあるmessageのフレームワークは不要なので削除する？
- [LANGUAGE_CODE](https://docs.djangoproject.com/en/5.1/ref/settings/#std-setting-LANGUAGE_CODE)は日本語にする？

**Comment by habvi** [comment]
-  DB を postgresql にする
   - 一応別タスクにしてありますが、一緒でもどちらでも大丈夫だと思います
- `INSTALLED_APPS`
  - Django に限らずフレームワークが自動生成した部分って、全部見て不要なものを消すっていう操作をしない気がしていて、特に問題が発生しないようなら消す必要はないのかな？と思ったりしています (確かに frontend に関する部分は不要かもですが、Django のデフォルトで管理サイトなどもあるので何に使われるかまだ不明な気がします)
- `LANGUAGE_CODE`
  - 今のところ日本語はコード内に書かない気がするので不要かも…？管理サイトが日本語になる、とかはありそうですね

他は特に初期設定必要なところは今のところないかと思っています！

**Comment by ayase-mstk** [comment]
おそらくdjango rest frameworkを使うようになるとsettings.pyに追加する項目が増えると思いますが、現状はあげられている項目くらいだと思います。
DBに関してはDB containerの設定もあるので別PRで行った方がいいかもしれません。

**Comment by s1-haya** [comment]
ご意見ありがとうございます。

結論
- DBはcontainerの設定もあるので別プルリク
- `INSTALLED_APPS`は問題があれば変更
- `LANGUAGE_CODE`はどっちでもいいから不要

**Comment by habvi** [comment]
すみません、まだ間に合えばで全然大丈夫なのですが、[この PR レビュー](https://github.com/42-pong/42-pong/pull/41#pullrequestreview-2426173459) で coderabbit が教えてくれたものがあり、
`SECRET_KEY` を環境変数から取得するようにする、というのも初期設定としてできそうです
全然別 Issue でついでにするとかでも大丈夫なので MUST ではないです！

> Secure the Django settings for production use.

> The SECRET_KEY should not be hardcoded in the settings file. Consider:
> 
> Using environment variables
> Using django-environ or python-dotenv
> Never committing the actual secret key to version control
> 
> Example implementation:
> 

> 
> ```py
> import os
> from pathlib import Path
> 
> # Use environment variables for sensitive settings
> SECRET_KEY = os.environ.get('DJANGO_SECRET_KEY', 'your-default-secret-key-for-development')
> ```

**Reply by s1-haya** [reply]
ありがとうござます！追加しときます！

**Reply by s1-haya** [reply]
すみません！ python-dotenv `か`django-envirsion`どっちにするかで依存関係が変わるため話してから設定しようと思うので別タスクにします

**Reply by habvi** [reply]
了解です！ありがとうございます (postgres に変更する PR で python-dotenv を install してしまった)
